testing/trino-product-tests/src/main/java/io/trino/tests/product/iceberg/TestIcebergSparkCompatibility.java

```diff
@@ -183,10 +192,10 @@ public void testTrinoReadingSparkData(StorageFormat storageFormat, int specVersi
         onSpark().executeQuery("DROP TABLE " + sparkTableName);
     }
 
-    @Test(groups = {ICEBERG, PROFILE_SPECIFIC_TESTS}, dataProvider = "testSparkReadingTrinoDataDataProvider")
+    @Test(groups = {ICEBERG, PROFILE_SPECIFIC_TESTS, ICEBERG_REST}, dataProvider = "testSparkReadingTrinoDataDataProvider")
```

Comment:
I am not sure yet the `TestIcebergSparkCompatibility` is the right test to run to cover Trino REST Catalog.

We should assume that Spark+Iceberg+HMS behaves same as Spark+Iceberg+REST, so we don't need to re-test Trino behavior vs Spark for REST, we already did that with HMS.

Having _a product test_ is definitely good. The product tests are the only thing that exercises all aspects of the deployment, including potential classloader issues.

So my preference would be 

- some simple product test for ensuring things work end-to-end, incuding potential packaging & classloader issues
- a test class extending from `BaseIcebergConnectorSmokeTest`, which would exercise all Trino operations is a more lightweight fashion.
  - added bonus is that this would be directly runnable from IDE

Reply:
Since we already have the full product test up and running, I think it would be good to keep it and there are differences between the REST Catalog and HMS that we would want to know about if there is a behavioral drift.

I also added the smoke tests as you suggested and validated they run cleanly through the IDE.